s4e-web/src/app/views/map-view/state/product/product.service.ts

```diff
@@ -37,21 +50,46 @@ export class ProductService {
       .subscribe((data) => this.store.set(data));
   }
 
+  getLastAvailableScene() {
+    const product = this.query.getActive();
+
+    if (product == null) {
+      return;
+    }
+
+    this.store.ui.update(product.id, state => ({...state, isLoading: true}));
+    this.http.get<MostRecentScene>(`${environment.apiPrefixV1}/products/${product.id}/scenes/most-recent`,
+      {params: {timeZone: timezone()}}
+    ).pipe(
+      finalize(() => this.store.ui.update(product.id, state => ({...state, isLoading: false})))
+    ).subscribe((data: MostRecentScene) => {
+      if (data.sceneId == null) {
+        this._notificationService.addGeneral({type: 'info', content: 'Ten produkt nie posiada jeszcze scen'});
+        return;
+      }
+
+      this.setSelectedDate(data.timestamp);
+      this.sceneService.get(product, data.timestamp.substr(0, 10));
+      this.sceneService.setActive(data.sceneId);
+    });
+  }
+
   setActive(productId: number | null) {
     this.store.ui.update({isLoading: false});
 
     if (productId != null) {
       this.store.ui.update(productId, {isLoading: true});
       const product = this.query.getEntity(productId);
       this.getSingle$(product)
-        .pipe(finalize(() => this.store.ui.update(productId, {isLoading: false})))
-        .subscribe((product) => {
-          applyTransaction(() => {
-            this.store.update(state => ({...state, ui: {...state.ui, loadedMonths: [], availableDays: []}}));
-            this.store.setActive(productId);
-          });
-          this.getAvailableDays();
+        .pipe(
+          finalize(() => this.store.ui.update(productId, {isLoading: false}))
```

Comment:
Why haven't you used handleHttpRequest$ operator? Why do you not handle errors in this pipe?